### PR TITLE
[PC-10625] fix report offer button

### PR DESCRIPTION
**Pull request by marionnousvalentin**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-10625

## Checklist

I have:

- [ ] Made sure the title of my PR follows the convention `[PC-10625] <summary>`.
- [ ] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Written **documentation on Notion** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |

     |         
image        |                          |          Screenshot_20210906-141530             |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

[PC-10625]: https://passculture.atlassian.net/browse/PC-10625

### CODE REVIEW

File: src/ui/components/modals/AppModal.tsx [hunk 1]
```diff
@@ -81,7 +82,9 @@ export const AppModal: FunctionComponent<Props> = ({
         isVisible={visible}
         onBackdropPress={handleOnBackdropPress()}
         style={[styles.container, styles.topOffset]}
-        testID="modal">
+        testID="modal"
+        deviceHeight={windowHeight}
```
Comment (emmajmn) [inline src/ui/components/modals/AppModal.tsx:84]: je comprends pas bien ce que ça permet ?

File: src/ui/components/modals/AppModal.tsx [hunk 2]
```diff
@@ -28,6 +27,20 @@ interface Props extends ModalStyles {
 
 const webcss = `div[aria-modal="true"] { align-items: center }`
 
+const defaultModalStyles = {
+  spacing: getSpacing(1),
+  height: getSpacing(112),
+  maxWidth: getSpacing(125),
```
Comment (emmajmn) [inline src/ui/components/modals/AppModal.tsx]: d'ou viennent ces valeurs ?

File: src/ui/components/modals/AppModal.tsx [hunk 3]
```diff
@@ -60,55 +60,54 @@ export const AppModal: FunctionComponent<Props> = ({
...
-            alignItems: 'center',
-            width: '100%',
-            borderTopStartRadius: getSpacing(4),
-            borderTopEndRadius: getSpacing(4),
-            padding: getSpacing(6),
-          },
-        }),
-      [modalLayout, spacing, modalHeight, modalMaxWidth]
-    )
+  const { layout: modalLayout, spacing, height: modalHeight, maxWidth: modalMaxWidth } = {
```
Comment (emmajmn) [inline src/ui/components/modals/AppModal.tsx]: je comprends pas bien cette refacto

**Review by emmajmn — APPROVED** [review]

File: src/ui/components/modals/AppModal.tsx (hunk 2 above)
Comment (marionnousvalentin) [reply src/ui/components/modals/AppModal.tsx]: de src/ui/components/modals/useModalStyles.ts, j'ai copié exactement (refacto)

File: src/ui/components/modals/AppModal.tsx (hunk 3 above)
Comment (marionnousvalentin) [reply src/ui/components/modals/AppModal.tsx]: plutot que d'avoir une fonction qui te renvoie un style, c'est direcctemen une const. C'est ensuite plus simple de passer le style en styleComponent

**Review by emmajmn — APPROVED** [review]

**Pull request merged**